### PR TITLE
Issue-6671: Fixed the order of plugins.

### PR DESCRIPTION
**Pull request by osho-20**
1. Why is this pull request needed and what does it do?
--> Fixed the issue #6671

2. Which issues (if any) are related?
--> Issue no #6671

3. Which documentation changes (if any) need to be made?
--> None

4. Does this introduce a backward incompatible change or deprecation?
--> No

### CODE REVIEW

**Comment by osho-20** [comment]
@miekg @chrisohaver @johnbelamaric @yongtang @stp-ip can you please review the code related to the bug issue.

**Review by chrisohaver — COMMENTED** [review]
can you use the existing Directives variable? https://github.com/coredns/coredns/blob/master/core/dnsserver/zdirectives.go

**Comment by osho-20** [comment]
surely will do it @chrisohaver

**Comment by osho-20** [comment]
@chrisohaver kindly review it

**Review by chrisohaver — DISMISSED** [review]

**Comment by chrisohaver** [comment]
Looks like the failed tests are related to the change. PTAL.

**Comment by osho-20** [comment]
Sure

**Review by chrisohaver — APPROVED** [review]

File: test/server.go [hunk 1]
```diff
@@ -40,7 +40,7 @@ func CoreDNSServerPorts(i *caddy.Instance, k int) (udp, tcp string) {
 	if t != nil {
 		tcp = t.String()
 	}
-	return
```
Comment (chrisohaver) [inline test/server.go]: Why this change?

Comment (osho-20) [reply test/server.go]: it must return udp and tcp right?

Comment (osho-20) [reply test/server.go]: Also this test case is running in my local but failing on github, just checking everything where a memory error or nil pointer error can occur.

Comment (chrisohaver) [reply test/server.go]: go doesn’t require listing the return variables if they are named in the signature. Also, it’s unrelated to plugin order, so doesn’t belong in this PR.

Comment (osho-20) [reply test/server.go]: okk will remove this now.

Comment (osho-20) [reply test/server.go]: this test is running successfully on my local and failing on github any specific reason, if you know else i will figure it out.

**Comment by osho-20** [comment]
@chrisohaver how to handle the permission, the error is
corefile_test.go:20: Expected no error, but got Listen: listen tcp :53: bind: permission denied

**Comment by osho-20** [comment]
can we merge this, if the changes are okay? @chrisohaver

**Comment by chrisohaver** [comment]
@osho-20, I think this looks good. Please resolve the conflicts.  I think the e2e test failure is a flake unrelated to your changes.  Hopefully it will pass when it re-runs after you resolve conflicts.

**Comment by osho-20** [comment]
Have resolved merge conflicts.

**Comment by osho-20** [comment]
can we merge this request
